lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -721,7 +721,7 @@ public function load(array $criteria, $entity = null, $assoc = null, array $hint
         $hydrator = $this->em->newHydrator($this->currentPersisterContext->selectJoinSql ? Query::HYDRATE_OBJECT : Query::HYDRATE_SIMPLEOBJECT);
         $entities = $hydrator->hydrateAll($stmt, $this->currentPersisterContext->rsm, $hints);
 
-        return $entities ? $entities[0] : null;
+        return $entities && isset($entities[0]) ? $entities[0] : null;
```

Comment:
Not for the normal setup, because in those cases the first part of the expression would still evaluate to false and the second part could cause a notice if the first part is missing.